Desktop: Persistent display config for each user session

kajusnau opened the pull request:
<!--
    Copyright 2023 TII (SSRC) and the Ghaf contributors
    SPDX-License-Identifier: CC-BY-SA-4.0
-->

## Description of changes

1. **Persistent display configuration per user session:**  
   - Saves a simple display configuration whenever the display setup changes.  
   - Stores enabled/disabled state, position, transform, and scale for each display.  
   - Default polling interval set to 1 minute.  
   - If a saved configuration is present, it is applied automatically on user login.

Notes:
- In rare cases, the display config may be lost on logout or when the session ends.  
  This can happen if the polling check happens within <1s time window as the session terminating.

<!--
Summary of the proposed changes in the PR description in your own words. For dependency updates, please link to the changelog.
-->

## Checklist for things done

<!-- Please check, [X], to all that applies. Leave [ ] if an item does not apply but you have considered the check list item. Note that all of these are not hard requirements. They serve information to reviewers. When you fill the checklist, you indicate to reviewers you appreciate their work. -->

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [x] Change can be updated with `nixos-rebuild ... switch`

<!-- Additional description of omitted [ ] items if not obvious. -->

## Instructions for Testing

- [ ] List all targets that this applies to:
- [x] Is this a new feature
  - [x] List the test steps to verify:
  - Login and adjust the display setup, preferably with 2 or more displays connected, focusing on these settings:
    - Positioning
    - Scaling
    - Enabled/Disabled state (optional)
  - Wait at least 1min before logging out or rebooting.
  - Verify your display config is re-applied upon login.
- [ ] If it is an improvement how does it impact existing functionality?

milva-unikie commented:
**Tested on Lenovo-X1** (`nixos-rebuild switch`)

Issues:
- Can't disable external screen, it comes back after a couple of seconds

Working:
- Configuration is saved
- Saved configuration is applied after logout/login and reboot
- Test-automation passes

kajusnau commented:
Made some adjustments, fixed a bug where the config application would fail and prevent auto-scaling from running.
Added transform state saving to the config.
No states (scale, position, transform) are saved when display is off (wlr-randr limitation).

milva-unikie commented:
**Tested on Lenovo-X1** (`nixos-rebuild switch`)
- External screen stays disabled (tested with a different screen than originally)
- Saved configuration now also includes the screen rotation
- Test-automation passes
